### PR TITLE
HPCC-19106 Secure socket avail_read() fix for when peer closes socket

### PR DESCRIPTION
**Pull request by mckellyln**
Signed-off-by: Mark Kelly <mark.kelly@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:

QA suite and additional ssl testsocket to Roxie tests pass as expected.

### CODE REVIEW

**Comment by mckellyln** [comment]
@mayx please review and report if your tests work ok with this branch

**Review by mayx — APPROVED** [review]
@mckellyln change looks fine, worked with my tests.

**Comment by mckellyln** [comment]
I propose then we go with this PR until a redesign/refactoring of the notifySelected msg loop handling code.
@afishbeck, @jakesmith - thoughts ?

File: system/security/securesocket/securesocket.cpp [hunk 1]
```diff
@@ -326,7 +327,31 @@ class CSecureSocket : implements ISecureSocket, public CInterface
...
+            pending = SSL_peek(m_ssl, c, 1);
+            // 0 almost always means socket was closed
+            if (pending == 0)
+                return 0;
+            if (pending > 0)
+                return SSL_pending(m_ssl);
+            // pending < 0 : TODO should handle SSL_ERROR_WANT_READ/WRITE error
+            if (m_loglevel >= SSLogNormal)
+            {
+                int ret = SSL_get_error(m_ssl, pending);
```
Comment (afishbeck) [inline system/security/securesocket/securesocket.cpp]: Am I missing something?  You don't seem to be using anything from this call.  I don't think ERR_get_error() requires SSL_get_error() to have been called?

Comment (mckellyln) [reply system/security/securesocket/securesocket.cpp]: es, thanks.  I meant to log the ret error.  I'll fix.

Comment (mckellyln) [reply system/security/securesocket/securesocket.cpp]: added ret to log.

**Review by afishbeck — APPROVED** [review]
@mckellyln looks good, especially as an interim fix until your larger changes can be put into place.

**Pull request merged**